src/routes/GroupDetail/GroupDetail.js

```diff
@@ -86,25 +87,24 @@ export class UnwrappedGroupDetail extends Component {
     const {
       currentUser,
       group,
+      history,
       isAboutCurrentGroup,
       isMember,
       location,
       moderators,
-      onClose,
       pending,
       routeParams
     } = this.props
+    const fullPage = !getRouteParam('detailGroupSlug', {}, this.props)
 
     if (!group && !pending) return <NotFound />
     if (pending) return <Loading />
 
-    const fullPage = !onClose
-
     return (
       <div className={cx({ [g.group]: true, [g.fullPage]: fullPage, [g.isAboutCurrentGroup]: isAboutCurrentGroup })}>
         <div styleName='g.groupDetailHeader' style={{ backgroundImage: `url(${group.bannerUrl || DEFAULT_BANNER})` }}>
-          {onClose && (
-            <a styleName='g.close' onClick={onClose}><Icon name='Ex' /></a>
+          {!fullPage && (
+            <a styleName='g.close' onClick={history.goBack}><Icon name='Ex' /></a>
```

Comment:
hmm, but what if there's no page to go back to? what if they loaded this page directly? or the page to go back to

Reply:
You're entirely correct. I think I was trying to ignore that case. I have updated now to cover that case and tested it for all the places this routes matches: https://github.com/Hylozoic/hylo-evo/pull/1353/files#diff-a9d11da53b47e893e82fb72c59e6020964f5d124e484f9dbd4213a3183cc30c8R396-R402

Should be good to go.

Reply:
I think the root issue here was potentially just that the `history.push` that was happening was via the `connect-react-router` API which may not have been populating the browser history object appropriately. That's my suspicion, and either way I consider that way of navigating legacy and we should use the React Router provided `history` directly. Whatever the case what is here is now doing what I need in the App.

Reply:
Bonus note: I created this initially and it worked fine, BUT MapExplorer initial load includes a redirect from the Stream somehow so in that case going back would from a first-page-load of the "About us" would send the user back to the Stream instead of the Group Detail. Hurumph. I'd like to stay as close to browser native stuff as possible (in avoidance of duplicating state management up the stack)... we were so close :) 

```
import { useHistory } from 'react-router-dom'

export default function useSafeGoBack (defaultPath = '/') {
  const history = useHistory()

  return () => {
    if (document.referrer.startsWith(document.location.origin)) {
      history.goBack()
    } else {
      history.push(defaultPath)
    }
  }
}
``